pkg/api/db.go

```diff
@@ -66,50 +66,34 @@ func CreateVolumeDBEntry(ctx *c.Context, in *model.VolumeSpec) (*model.VolumeSpe
 	if in.CreatedAt == "" {
 		in.CreatedAt = time.Now().Format(constants.TimeFormat)
 	}
-	vol := &model.VolumeSpec{
-		BaseModel: &model.BaseModel{
-			Id:        in.Id,
-			CreatedAt: in.CreatedAt,
-		},
-		UserId:            ctx.UserId,
-		Name:              in.Name,
-		Description:       in.Description,
-		ProfileId:         in.ProfileId,
-		Size:              in.Size,
-		AvailabilityZone:  in.AvailabilityZone,
-		Status:            model.VolumeCreating,
-		SnapshotId:        in.SnapshotId,
-		SnapshotFromCloud: in.SnapshotFromCloud,
-	}
-	result, err := db.C.CreateVolume(ctx, vol)
-	if err != nil {
-		log.Error("When add volume to db:", err)
-		return nil, err
-	}
 
-	return result, nil
+	in.Status = model.VolumeCreating
+	// Store the volume data into database.
+	return db.C.CreateVolume(ctx, in)
 }
 
-func ExtendVolumeDBEntry(ctx *c.Context, volID string) (*model.VolumeSpec, error) {
+func ExtendVolumeDBEntry(ctx *c.Context, volID string, newSize int64) (*model.VolumeSpec, error) {
```

Comment:
Why newSize has to be added now?  How did it work before without the newSize?

Reply:
As u can see from below, I added some validation logic in this method.

Reply:
As u can see from below, I added some validation logic in this method.

Reply:
Some comments should be added to explain why volume.size is not changed to the new size.  If you are just validating, but not really changing the size, explain when the size is really changed.

Also should this validation be done earlier in the API layer, before entering ExtendVolumeDBEntry?

Reply:
Currently all of validation logic (status, dependency, etc) are implemented in ***DBEntry method, so I just kept that style and one of benefits for that is good for unit test.